maps/BattleTower1F.asm

```diff
@@ -7,8 +7,8 @@
 
 BattleTower1F_MapScripts:
 	def_scene_scripts
-	scene_script .Scene0 ; SCENE_DEFAULT
-	scene_script .Scene1 ; SCENE_FINISHED
+	scene_script .Scene0, SCENE_BATTLETOWER1F_CHECKSTATE
+	scene_script .Scene1, SCENE_BATTLETOWER1F_NOTHING
```

Comment:
I personally prefer `SCENE_MAPNAME_DEFAULT` for the initially set default scenes.. as this would match up better with what i'm doing in pokered's PR. Also my PR uses `SCRIPT_MAPNAME_NOOP` for the dummy scripts; perhaps something similar for the nothing scenes?

Just thinking about consistency.

Reply:
I chose not to use `SCENE_*_DEFAULT` because it doesn't say what the default scene is for. Sometimes it's for `NOTHING`, sometimes for a one-time event like meeting an NPC that gets permanently switched to a `NOTHING` scene, sometimes it might switch back to the default scene later.

pokecrystal has used `SCENE_NOTHING` for do-nothing scenes for a while; maybe pokered should use `NOTHING` instead of `NOOP`? It's longer, but also an actual word. Thoughts @dannye @mid-kid ?

Reply:
I think that not using `SCENE_*_DEFAULT` is fine then.. I'd still prefer consistency for whatever we decide to go with for the Do nothing scenes/scripts. 

Also another note... Should we rename the scripts as well? Like:
```asm
	scene_script .CheckStateScene, SCENE_BATTLETOWER1F_CHECKSTATE
	scene_script .NothingScene,    SCENE_BATTLETOWER1F_NOTHING
```

Reply:
I don't know enough about gen 2 to really know whether `SCENE_*_DEFAULT` is or is not appropriate.

At least in gen 1, a lot of maps' "default" script handles multiple things and executes every frame indefinitely so "default" is a perfectly adequate name imo (doesn't mean it *always* has to be named like this, if a better name exists on a case-by-case basis). It's also nice to know - at a glace - that the "default" script is the one that will execute on each new save until the map's script is explicitly updated.
If some part of this (with regard to how the scripting engines work) doesn't apply to gen 2, then I think it's okay if pokered and pokecrystal are a bit inconsistent.

I like `NOOP` more than `NOTHING`.

> It's longer, but also an actual word.

It may not be a dictionary word, but it's not like it's just made up either. Not really a relevant point in my opinion.

Reply:
I don't mind having a consistent `<MAPNAME>_NOOP`, DEFAULT, or NOTHING or similar, but the value of the "nothing" scene isn't consistent across every map. Like rangi mentioned, scene 0 is often a one-time event.

Reply:
I'll change `NOTHING` to `NOOP` for consistency then.